pkg/kn/commands/source/apiserver/create_flag.go

```diff
@@ -0,0 +1,77 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	sources_v1alpha1 "knative.dev/eventing/pkg/apis/sources/v1alpha1"
+)
+
+const (
+	ApiVersionSplitChar           = ":"
+	ApiServerSourceArraySplitChar = ","
+	DefaultApiVersion             = "v1"
+)
+
+// ApiServerSourceUpdateFlags are flags for create and update a ApiServerSource
+type ApiServerSourceUpdateFlags struct {
+	ServiceAccountName string
+	Mode               string
+	Resources          string
+}
+
+// GetApiServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
+func (f *ApiServerSourceUpdateFlags) GetApiServerResourceArray() []sources_v1alpha1.ApiServerResource {
+	var resourceList []sources_v1alpha1.ApiServerResource
+	for _, r := range strings.Split(f.Resources, ApiServerSourceArraySplitChar) {
+		version, kind, controller := getValidResource(r)
+		resourceRef := sources_v1alpha1.ApiServerResource{
+			APIVersion: version,
+			Kind:       kind,
+			Controller: controller,
+		}
+		resourceList = append(resourceList, resourceRef)
+	}
+	return resourceList
+}
+
+func getValidResource(resource string) (string, string, bool) {
+	var version = DefaultApiVersion // v1 as default
```

Comment:
would call it `apiGroupVersion` as this is supposed be the api-group + version combined (like in in `apps/v1` or `serving.knative.dev/v1alpha`)

Reply:
Also, IMO the version should be mandatory. Having `v1` as a default doesn't not make sense except for the most basice objects which does not carry an API group.